atmel-samd/mphalport.c

```diff
@@ -210,3 +210,35 @@ void mp_hal_delay_us(mp_uint_t delay) {
     }
     delay_us(delay);
 }
+
+// Global millisecond tick count (driven by SysTick interrupt handler).
+volatile uint32_t systick_ticks_ms = 0;
+
+void SysTick_Handler(void) {
+    // SysTick interrupt handler called when the SysTick timer reaches zero
+    // (every millisecond).
+    systick_ticks_ms += 1;
+    // Keep the counter within the range of 31 bit uint values since that's the
+    // max value for micropython 'small' ints.
+    systick_ticks_ms = systick_ticks_ms > 2147483647L ? 0 : systick_ticks_ms;
```

Comment:
Can you switch this constant to either a bit shift (1 << 31) or hex? I think it'd be more readable that way.